src/PsrContainerFakerProvider.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CalebDW\Fakerstan;
+
+use Faker\Generator;
+use Psr\Container\ContainerInterface;
+
+final class PsrContainerFakerProvider implements FakerProvider
+{
+    private ?Generator $generatorFromContainer = null;
+
+    public function __construct(
+        private string $phpContainerPath,
+        private ?string $setsVariable,
+        private string $containerFakerId,
+    ) {
+    }
+
+    public function getFaker(): Generator
+    {
+        if (is_null($this->generatorFromContainer)) {
+            $this->generatorFromContainer = $this->getGeneratorFromContainer();
+        }
+
+        return $this->generatorFromContainer;
+    }
+
+    private function getGeneratorFromContainer(): Generator
+    {
+        if (! is_readable($this->phpContainerPath)) {
+            throw new \RuntimeException('Could not read container PHP file');
+        }
+
+        // Include the container file and store the return value, just in case.
+        $maybeContainer = require_once $this->phpContainerPath;
+
+        // If the code in that file returns nothing, the require_once will return 1.
```

Comment:
I appreciate all the comments, but the code is pretty straightforward---please remove them